test/regridder_tests.jl

```diff
@@ -314,3 +315,54 @@ for FT in (Float32, Float64)
         end
     end
 end
+# test dataset truncation 
+@testset "test dataset truncation" begin
+    # Get the original dataset set up 
+    include(joinpath(pkgdir(ClimaCoupler), "artifacts", "artifact_funcs.jl"))
+    sst_data_all = joinpath(sst_dataset_path(), "sst.nc")
+    ds = NCDataset(sst_data_all, "r")
+    dates = ds["time"][:]
+    first_date = dates[1]
+    last_date = last(dates)
+
+    # values for the truncations 
+    time_start = 0.0
+    time_end = 1.728e6
+    # tests five cases: 
+    # 1. both start and end are earlier than in datafile 
+    # 2. start is early, end is within datafine 
+    # 3. both start and end within datafile 
+    # 4. start is in datafile end is not 
+    # 5. both start and end are later than in datafile
```

Comment:
Nice tests! Could we also add a test that the sst data is correctly indexed (as well as the time)? I see from running this locally that sst has the correct dimensions, but it could be good to test that `original_sst[index_of_start_date] == truncated_sst[1]`

Reply:
Would this need my function to return the start index? It is all calculated in the function so I don't think there would be a way to know what the index_of_start_date is other than just looking at the index of the first value in trucated_sst defeating the purpose of the test